docs/architecture/adr-100-data-companion-push-api.md

```diff
@@ -0,0 +1,441 @@
+      over providing similar guarantees to a scalable RPC interface (especially
+      since we know this does not scale, and if such a job could just be
+      outsourced). If we have a single consumer, we can easily automatically prune
+      data that was already sent. We could overcome this by limiting the API to a
+      single consumer and single event window.
+
+   Making the requisite modifications to ADR-075 basically converges on the
+   solution outlined in this ADR, except that ADR-075 publishes data via JSON over
+   HTTP while this solution proposes gRPC.
+
+2. Another alternative is proposed in [ADR 101][adr-101], which has the same
+   requirements as this proposed approach, except it implements a "pull" model
+   instead.
+
+3. We could pick a database that provides real-time notifications of inserts to
+   consumers and just dump the requisite data into that database. Consumers
+   would then need to connect to that database, listen for updates, and then
+   transform/process the data as needed.
+
+   This approach could also inevitably lead to feature sprawl in people wanting
+   us to support multiple different databases. It is also not clear which
+   database would be the best option here, as that would most likely be
+   influenced by integrators' use cases.
+
+   Beyond this, if we exclusively support one database but integrators needed
+   another for their use case, they would be forced to perform some kind of
+   continuous ETL operation to transfer the data from the one database to the
+   other, potentially doubling their storage capacity requirements and therefore
+   their storage-related costs. This would also increase pruning complexity.
+
+## Decision
+
+> TODO(thane)
+
+## Detailed Design
+
+### Use Cases
+
+1. Integrators could provide standalone RPC services that would be able to scale
+   independently of the Tendermint node. They would also be able to present the
+   data in whatever format and by whichever protocol they want.
+2. Block explorers could make use of this data to provide real-time information
+   about the blockchain.
+3. IBC relayer nodes could use this data to filter and store only the data they
+   need to facilitate relaying without putting additional pressure on the
+   consensus engine (until such time that a decision is made on whether to
+   continue providing event data from Tendermint).
+
+### Requirements
+
+1. A node _must_ support at most one data companion.
+
+2. All or part of the following data _must_ be obtainable by the companion, and
+   as close to real-time as possible:
+   1. Committed block data
+   2. `FinalizeBlockResponse` data, but only for committed blocks
+
+3. The companion _must_ be able to establish the earliest height for which the
+   node has all of the requisite data.
+
+4. The API _must_ be (or be able to be) appropriately shielded from untrusted
+   consumers and abuse. Critical control facets of the API (e.g. those that
```

Comment:
Line 20 above says "single trusted external companion service". So I would clarify that the shielding we talk about here does not include a (trusted) companion, but unauthorized actors.